File: library/src/main/java/com/pengrad/telegrambot/model/stars/partner/TransactionPartnerOther.java
package com.pengrad.telegrambot.model.stars.partner;

import java.util.Objects;

public class TransactionPartnerOther extends TransactionPartner {

    public final static String TYPE = "other";

    public TransactionPartnerOther() {
        super(TYPE);
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TransactionPartnerOther that = (TransactionPartnerOther) o;
        return Objects.equals(type(), that.type());
    }

    @Override
    public int hashCode() {
        return Objects.hash(type());
    }

    @Override
    public String toString() {
        return "TransactionPartnerOther{" +
                "type='" + type() + "\'" +
                '}';
    }

}

File: library/src/main/java/com/pengrad/telegrambot/model/stars/partner/TransactionPartnerOther.kt
package com.pengrad.telegrambot.model.stars.partner

import com.pengrad.telegrambot.model.stars.partner.TransactionPartnerType.OTHER

class TransactionPartnerOther : TransactionPartner {

    override val type: String
        get() = OTHER

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TransactionPartnerOther

        return type == other.type
    }

    override fun hashCode(): Int {
        return type.hashCode()
    }

    override fun toString(): String {
        return "TransactionPartnerOther(type='$type')"
    }
}